lib/token.js

```diff
@@ -12,8 +12,37 @@ function getWellKnown(sdk) {
   return http.get(sdk, sdk.options.url + '/.well-known/openid-configuration');
 }
 
-function validateClaims(sdk, claims, aud) {
-  var iss = sdk.options.url;
+function getAuthServerUrl(sdk, authServer) {
+  authServer = authServer || sdk.options.authServer;
+  if (authServer) {
+    if (/^https?:/.test(authServer)) {
+      return authServer;
+    }
+    return sdk.options.url + '/oauth2/' + authServer;
+  }
+}
+
+var isAuthorizationServer = new RegExp('^https?://.*?/oauth2/.+');
+function issuerToAuthorizationUrl(issuer) {
+  if (isAuthorizationServer.test(issuer)) {
+    return issuer;
+  }
+  return issuer + '/oauth2';
+}
+
+function getIssuer(sdk, authServer) {
```

Comment:
To make this a bit clearer, can we use these three names (with associated functions):
- issuer: getIssuer -> function that either takes in function param for issuer, initialize param for issuer, or defaults to options.uri
- authorizeBaseUrl: getAuthorizeBaseUrl -> calls getIssuer and appends /oauth2 if it is the default case of options.uri (i.e. does not have /oauth2 there)
- authorizeUrl: getAuthorizeUrl -> adds the /v1/authorize part to authorizeBaseUrl